### PR TITLE
fix(module-order): Removes async module addition in favour of synchronous array replacement

### LINKED ISSUE #353: Moving module doesn't move it
## Problem
I have a preset and I want to move a module (in this case `Pixelate`) into another position in my layer. When I do this and save my preset, the module is saved in another position inside the preset-json, but loading my preset always puts the module in the position it was before I moved it. 

## Steps to reproduce
* Load my preset [default.zip](https://github.com/vcync/modV/files/3591734/default.zip)
* Move the pixelate module to the end of the layer
* Save my preset
* Reload modV
* Load my preset
* See that pixelate is at the same position as before

## What I did so far
* I removed the media folder and copied everything from the backup into the `default` project
* I restarted modV on the CLI, still the same problem

### PR DESCRIPTION
**Pull request by 2xAA**
Potentially fixes #353

### CODE REVIEW

File: src/components/ActiveModule/OpacityControl.vue [hunk 1]
```diff
@@ -24,7 +24,10 @@ export default {
     alpha: {
       get() {
         if (!this.moduleName) return 0
-        return this.$store.state.modVModules.active[this.moduleName].meta.alpha
+        return (
+          this.$store.state.modVModules.active[this.moduleName] &&
+          this.$store.state.modVModules.active[this.moduleName].meta.alpha
```
Comment (TimPietrusky) [inline src/components/ActiveModule/OpacityControl.vue:29]: Why is this change needed?

File: src/components/ActiveModule/index.vue [hunk 2]
```diff
@@ -245,8 +245,10 @@ export default {
     enabled: {
       get() {
         if (!this.moduleName) return false
-        return this.$store.state.modVModules.active[this.moduleName].meta
-          .enabled
+        return (
+          this.$store.state.modVModules.active[this.moduleName] &&
+          this.$store.state.modVModules.active[this.moduleName].meta.enabled
```
Comment (TimPietrusky) [inline src/components/ActiveModule/index.vue:250]: And this change?

**Review by TimPietrusky — CHANGES_REQUESTED** [review]
I just have two questions!

File: src/components/ActiveModule/OpacityControl.vue (hunk 1 above)
Comment (2xAA) [reply src/components/ActiveModule/OpacityControl.vue:29]: I think I was relying on async timing(/race condition) before the change to the module order, but I'm not 100% sure on this.

Both this and the other Vue file change produced errors after I made the module order change as they were trying to access data which is now not present when the component is created.

Tbh both of those Vue files need a complete rewrite, they're full of terrible code.

File: src/components/ActiveModule/index.vue (hunk 2 above)
Comment (2xAA) [reply src/components/ActiveModule/index.vue:250]: I think I was relying on async timing(/race condition) before the change to the module order, but I'm not 100% sure on this.

Both this and the other Vue file change produced errors after I made the module order change as they were trying to access data which is now not present when the component is created.

Tbh both of those Vue files need a complete rewrite, they're full of terrible code.

**Pull request merged**